### DISCUSSION TITLE
mongoose not working with vercel hosting

### DISCUSSION THREAD

**Discussion by youssefemad95** [vercel/community#2711] [Help]
### Summary

I created a very simple express app and I'm deploying it to vercel as a serverless function. It works great until I add this line to my code:
`const mongoose = require('mongoose');`

Everything works fine on localhost, but when I deploy my app even before I connect to my mongoDB database, I get this error:
500: INTERNAL_SERVER_ERROR
Code: FUNCTION_INVOCATION_FAILED
ID: cdg1::g7zlg-1685734348695-5e2d9ee57c90

Here's my code:
`require('dotenv').config();
const express = require('express');
const cors = require('cors');
const mongoose = require('mongoose');
const app = express();
app.use(cors());
const port = 3000;

app.get('/', (req, res) => {
    res.send("Hello World");
})

app.listen(port, (req, res) => {
    console.log('listening on port ' + port);
})

module.exports = app;`

If I remove this line `const mongoose = require('mongoose');` The deploymenet will work. Why is this happening?

### Steps to Reproduce

If I add this line: `const mongoose = require('mongoose');` the deployment won't work. When I remove it from the code, it works fine.

**Comment by amyegan** [comment]
Hi @youssefemad95. Do you see any additional error info in the [runtime logs](https://vercel.com/docs/concepts/observability/runtime-logs)?

**Reply by youssefemad95** [reply]
Hi @amyegan 
This shows in the log
RequestId: 7f6ce397-54c0-4a7d-81a5-2b819a564c31 Error: Runtime exited with error: exit status 1
Runtime.ExitError
